Framework/Algorithms/src/RunCombinationHelpers/SampleLogsBehaviour.cpp

```diff
@@ -375,7 +375,7 @@ std::shared_ptr<Property> SampleLogsBehaviour::addPropertyForList(
     const std::string &item, const std::string &value, MatrixWorkspace &ws) {
   std::shared_ptr<Property> returnProp;
 
-  const std::string unit = ws.getLog(item)->units();
+  const std::string &unit = ws.getLog(item)->units();
```

Comment:
Here I think we have the same issue as above: `unit` is a backup and making it a reference breaks the intended behavior.